### PR TITLE
Ticket 6: GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by hellodanielwilliams**
## Setup / config
* Adds comments.controllers and comments.models files

## Endpoint implementation
* Adds happy path tests for a GET on /api/articles/:article_id/comments and implements functionality to pass
* Adds 404 and 400 sad path tests for non-existent and non-valid article_id and passes them
* Adds another GET 200 test for cases where an article_id exists but has no associated comments, to serve an empty array.
* This is dealt with by creating a function to check if article exists in the articles model which rejects if not, then implementing a Promise.all in the comments controller that calls that function as well as the one to select comments 

## Other
* Updates endpoints.json with details for this endpoint

## Notes
* I added a 400 test for invalid path here which was already passing due to having already dealt with a 22P02 db error from the previous endpoint. Would best practice be to keep this test in or delete it as it was already passing?
* Improved destructuring in the tests for this endpoint based on something I learned from this morning's lecture. Will refactor the previous ones for consistency later

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -123,4 +123,35 @@ describe('/api/articles/:article_id', () => {
...
+            const { comments } = body
+            expect(comments.length).toBe(11)
+            comments.forEach((comment => {
+                const {comment_id, votes, created_at, author, body, article_id} = comment
+                expect(typeof comment_id).toBe('number')
+                expect(typeof votes).toBe('number')
+                expect(typeof created_at).toBe('string')
+                expect(typeof author).toBe('string')
+                expect(typeof body).toBe('string')
+                expect(typeof article_id).toBe('number')
```
Comment (nc-beth-mccormack) [inline __tests__/app.test.js:141]: You can be more specific here - do you expect this to be any number?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -123,4 +123,35 @@ describe('/api/articles/:article_id', () => {
...
+            }))
+        })
+    })
+    test('Get 200: responds with comments array ordered in date descending by default', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({ body }) => {
+            const { comments } = body
+            console.log(comments)
```
Comment (nc-beth-mccormack) [inline __tests__/app.test.js]: remove console.logs before committing

File: models/comments.models.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const db = require('../db/connection')
+
+exports.selectCommentsByArticleId = (article_id) => {
+    return db.query(`
+        SELECT * FROM comments
+        WHERE article_id = $1
+        ORDER BY created_at DESC
+    ;`, [article_id])
```
Comment (nc-beth-mccormack) [inline models/comments.models.js:8]: Nice job avoiding SQL injection✅

File: controllers/comments.controllers.js [hunk 4]
```diff
@@ -5,4 +5,5 @@ exports.getCommentsByArticleId = (req, res, next) => {
     return selectCommentsByArticleId(article_id).then((comments) => {
         res.status(200).send({ comments })
     })
+    .catch(next)
```
Comment (nc-beth-mccormack) [inline controllers/comments.controllers.js:11]: ✅

File: __tests__/app.test.js [hunk 5]
```diff
@@ -167,5 +167,13 @@ describe('/api/articles/:article_id/comments', () => {
             .then(({ body: { msg } }) => {
             expect(msg).toBe('Bad request')
         })
-    })  
+    })
+    test('GET 200: responds with an empty array if the article_id exists but has no comments', () => {
```
Comment (nc-beth-mccormack) [inline __tests__/app.test.js:169]: A good test to add✅

File: __tests__/app.test.js [hunk 6]
```diff
@@ -167,5 +167,13 @@ describe('/api/articles/:article_id/comments', () => {
...
+        return request(app)
+        .get('/api/articles/2/comments')
+        .expect(200)
+        .then(({ body: { comments } }) => {
+            expect(comments).toHaveLength(0)
```
Comment (nc-beth-mccormack) [inline __tests__/app.test.js:174]: It could also be suitable here to check the article id, to prove that its the correct article with no comments and not in fact an article which should have comments

**Review by nc-beth-mccormack — COMMENTED** [review]
I've left some comments 🙂
